docs/website/contents/for-developers/HardForks.md

```diff
@@ -0,0 +1,98 @@
+
+- The familiar and intuitive notion of era is useful for organization in general, and so the eras' names might appear within parameter names in configuration files, specification documents, announcements, etc.
+
+  However, the era transitions themselves should generally be considered as "just a big hard fork".
+  The foundational data is instead the protocol version, especially because not all hard forks are era transitions.
+  (See [this GitHub comment](https://github.com/IntersectMBO/ouroboros-consensus/issues/416#issuecomment-2669347315) for the details of how this lesson was learned.)
+
+- Some codecs reflect Cardano's specific era structure.
+  That's  not strictly necessary, but a tagged union approach has made it trivial to combine the era-specific codecs into a full Cardano codec.
+  Specifically, top-level codecs fundamentally branch on a simple number located near the beginning of the bytestring and dispatch accordingly to the corresponding era's codecs.
+
+- (TODO Confirm whether these tags dont affect any hashes. If they do, to what extent does it actually matter?)
+
+## Forecasting, forewarning
+
+Hard forks can affect the validity of block headers, and so must be sufficiently forecastable.
+For this reason, hard forks must not happen less than 36 hr (on mainnet) after the voting deadline, ie hard forks must have at least 36 hrs of forewarning --- just like other protocol parameter updates that affect headers (eg maximum block size).
+
+TODO how to argue that hard forks or era transitions needs more than that?
+
+[^updatable-protocol-parameters]: *Reference*.
+  See Section 5 "Protocol Parameters" of the "A Formal Specification of the Cardano Ledger, Deliverable SL-D5" document for more details.
+  Its latest revision can be accessed via the hyperlink in the "Formal Specification" column of the "Shelley" row of the documentation table at the top of the [README](https://github.com/IntersectMBO/cardano-ledger/blob/master/README.md) of the [`cardano-ledger` repository](https://github.com/IntersectMBO/cardano-ledger).
+
+[^change-decisions]: *Reference*.
+  Nowadays, these are almost always going to be (possibly indirectly) related to a Cardano Improvement Proposal (aka [CIP](https://cips.cardano.org/)).
+
+[^frame-rule]: *Clarification*.
+  There's no other technical reason for the community to increment the major protocol version.
+
+[^roadmap-disambiguation]: *Disambiguation*.
+  The [Cardano Roadmap](https://roadmap.cardano.org/en/) also defines some very broad "eras": Byron, Shelley, Goguen, Basho, and Voltaire.
+  These are more general than the eras that divide the actual evolution of the protocol rules and ledger rules within the node.
+  [CIP-0059](https://github.com/cardano-foundation/CIPs/blob/master/CIP-0059/feature-table.md) maintains a table showing the correspondence; it lists roadmap eras in the "Phase" column.
+
+[^accidentals]: *Clarification*.
+  Accidental hard forks would necessarily be intra-era hard forks.
```

Comment:
What is accidental hard fork? Are you talking about an uncontrolled hard fork due to divergence in validity of a block?
If that is the case then it can't be considered as an intra-era hard fork either, IMHO, because no protocol version update would happen. 

However, if you are talking about _accidental_ fork as it is described on wikipedia, then it is not even classified as a *hard* fork, it is part of the normal process of chain selection, which has a potential to span over era boundaries, as far as I understand it.

Reply:
Hmm. Yeah, I was taking the "intra-era" part too literally. It's a good point that what we call an _intra-era hard fork_ necessarily involves a major protocol version bump, and that wouldn't be accidental.